Documentation/network/clustermesh/mcsapi.rst

```diff
@@ -0,0 +1,165 @@
+multicluster plugin, for instance by executing the following command on each cluster:
+
+   .. code-block:: shell-session
+
+      kubectl get configmap -n kube-system coredns -o yaml | \
+         sed -E -e 's/^(\s*)kubernetes.*cluster\.local.*$/\1multicluster clusterset.local\n&/' | \
+         kubectl replace -f-
+
+And you can finally deploy the CoreDNS image you previously built. Doing so will
+also rollout the CoreDNS deployment and activate the Corefile change you previously made.
+
+Exporting a Service
+###################
+
+To export a service you should create a ServiceExport resource. As a result
+your Service will be exported to all clusters, provided that the Service
+Namespace is present on those clusters.
+
+   .. code-block:: yaml
+
+      apiVersion: multicluster.x-k8s.io/v1alpha1
+      kind: ServiceExport
+      metadata:
+         name: rebel-base
+
+In all the clusters and for each set of exported Services that have the same name and namespace,
+a ServiceImport resource will be automatically created. All the Endpoints from those exported Services
+with the same name and namespace will be merged and made globally available.
+
+An exported Service through MCS-API is available by default on the ``<svc>.<ns>.svc.clusterset.local`` domain.
+If you have defined any hostname (via a Statefulset for instance) on your pods
+each pods would also be available available through the ``<hostname>.<clustername>.<svc>.<ns>.svc.clusterset.local`` domain.
+
+   .. note::
+
+      The ``<clustername>.<svc>.<ns>.svc.clusterset.local`` domain that would allow
+      to get all the endpoints of a Service in a specific cluster is not allowed!
+
+      We recommend creating one service per cluster and/or region and exporting it
+      accordingly if you do want to have this kind of behavior, for instance
+      creating and exporting services ``mysvc-eu``, ``mysvc-us`` instead of only
+      one service. For more information checkout the `dedicated section <MCS_API_KEP_clusterdns>`_
+      in the MCS-API KEP explaining this behavior.
+
+.. _MCS_API_KEP_clusterdns: https://github.com/kubernetes/enhancements/blob/master/keps/sig-multicluster/1645-multi-cluster-services-api/README.md#not-allowing-cluster-specific-targeting-via-dns
+
+The ServiceImport also has a logic to merge different Service properties:
+
+- SessionAffinity
+- Ports (Union of the different ServiceExports)
+- Type (ClusterSetIP/Headless)
+
+If any conflict arise on any of those properties, the oldest ServiceExport will
+have precedence to resolve the conflict. This means that you should get a
+consistent behavior globally for the same set of exported Services that has
+the same name and namespace. If any conflicts arise you would be able to see
+details about it in the ServiceExport status Condition.
+
+
+Deploying a Simple Example Service using MCS-API
+------------------------------------------------
+
+1. In cluster 1, deploy:
+
+   .. code-block:: shell-session
```

Comment:
Same here, and below, about the need of using `parsed-literal` rather than `code-block`.